### ISSUE TITLE
Dropout Test phase, Weight scaling 

### ISSUE THREAD

**Issue by TullioM94** [rstudio/keras#1040]
I have a question for @jjallaire: 

I was not able to find any documentation regarding the functionality of dropout during test phase. 

In particular, If look at the issue: #358 it is not clear if the weight scaling usually applied during training with dropout, is implemented also in this case or not; I am asking this because if we allow for dropout also during test phase, we should not apply the weight scaling. 

Thank you for your help 

Best, 

Tullio

**Comment by henry090** [comment]
By default, Keras applies dropout only for training (if training = FALSE). However, if you enable ```training``` then it will be applied to test data as well. https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dropout

**Comment by TullioM94** [comment]
Hi! 

Yes, that was clear¬ my question was related to the weight scaling: 

Usually when we use dropout, at test phase we should scale the trained weights by p --> pW. In keras this is done by scaling the weights during training (1/p)W, so that no weight scaling should be applied at the test phase. 

However, when dropout is applied also in the test phase to generate stochastic forward passes, we should not scale the weights during the training phase, and it is not clear from the documentation if training = TRUE takes care of this aspect or not

**Comment by henry090** [comment]
I think you may find what you want here:
- https://github.com/keras-team/keras/issues/5357#issuecomment-282058994
- https://github.com/keras-team/keras/issues/9412
- https://stackoverflow.com/questions/38592943/is-the-keras-implementation-of-dropout-correct

Does it help?

**Comment by TullioM94** [comment]
Hi, 

Not really, it confirms what I was asking before and it shows (in the second point of the list) that some people asked the same question but they did not receive an answer.

**Comment by TullioM94** [comment]
@henry090 

I have looked at the source code fro dropout and I have the answer based also on what you sent: 

When dropout is implemented during training phase the weights are scaled as (1/p)W, and at test phase we use the non-scaled weights W. 

When we apply dropout at test phase we use both during training and test phase the scaled weights (1/p)W. 

Thus, in both cases we correctly implement the weight scaling. I was not able to understand as I thought that the weights scaling was implemented differently. 


Thank you for your help 


Tullio
